sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileScanBuilder.scala

```diff
@@ -57,6 +60,15 @@ abstract class FileScanBuilder(
     StructType(fields)
   }
 
+  def setFilters(pFilters: Seq[Expression], dFilters: Seq[Expression]): Unit = {
```

Comment:
should we make these two fields immutable and have a `withFilters` method?

Reply:
And +1 for keeping `withFilters` method name.

Reply:
Yes, +1 for keeping withFilters method name